spec/controllers_spec/posts_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+require 'factory_bot_rails'
+
+RSpec.describe 'Posts', type: :request do
+  before(:each) do
+    # Clean up the test database before each test
+    Post.destroy_all
+  end
+
+  describe 'GET /posts' do
+    it 'works! (now write some real specs)' do
+      get posts_path
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns all posts' do
+      expect(Post.count).to eq(0)
+    end
+
+    it 'renders the index template' do
+      get posts_path
+      expect(response).to render_template(:index)
+    end
```

Comment:
- [ ] Please write tests for the `show` action on the `PostsController`. These tests should at least do the following:
1. Ensure correct view template is rendered. E.g: `expect(response).to render_template(:show)`
2. Ensure the response body includes the correct placeholder text